### PR TITLE
Support NIST P-521 public keys

### PR DESCRIPTION
**Pull request by akeamc**
This PR adds support for the `ecdsa-sha2-nistp521` algorithm. The code is far from perfect and could use some polishing.

### CODE REVIEW

File: russh-keys/src/key.rs [hunk 1]
```diff
@@ -282,6 +301,31 @@ impl PublicKey {
...
+                    let f = reader.read_mpint().ok()?;
+                    let f = f.strip_prefix(&[0]).unwrap_or(f);
+                    let mut result = [0; FIELD_LEN];
+                    if f.len() > FIELD_LEN {
+                        return None;
+                    }
+                    #[allow(clippy::indexing_slicing)] // length is known
+                    result[FIELD_LEN - f.len()..].copy_from_slice(f);
+                    // Some(result.into())
+                    Some(GenericArray::clone_from_slice(&result)) // ew
```
Comment (akeamc) [inline russh-keys/src/key.rs:318]: GenericArray strangely does not implement `From<[u8; 66]>`. Any ideas?

Comment (Eugeny) [reply russh-keys/src/key.rs:318]: Yeah, `p521` is using an older version of `generic_arary` where `From<[T; N]>` is only implemented for N up to 64

**Comment by Eugeny** [comment]
Looking good so far, could you please also add a key parsing test?

**Comment by akeamc** [comment]
Sure!

**Comment by Eugeny** [comment]
Perfect! In the future I'd love to genericize the ECDSA code but for now it does the trick.

@all-contributors please add @akeamc for code

**Pull request merged**